agreement_legal/models/agreement.py

```diff
@@ -0,0 +1,412 @@
+    )
+    termination_date = fields.Date(
+        string="Termination Date",
+        tracking=True,
+        help="Date that the contract was terminated.",
+    )
+    reviewed_date = fields.Date(string="Reviewed Date", tracking=True)
+    reviewed_user_id = fields.Many2one("res.users", string="Reviewed By", tracking=True)
+    approved_date = fields.Date(string="Approved Date", tracking=True)
+    approved_user_id = fields.Many2one("res.users", string="Approved By", tracking=True)
+    currency_id = fields.Many2one("res.currency", string="Currency")
+    partner_id = fields.Many2one(
+        "res.partner",
+        string="Partner",
+        required=False,
+        copy=True,
+        help="The customer or vendor this agreement is related to.",
+    )
+    partner_contact_id = fields.Many2one(
+        "res.partner",
+        string="Partner Contact",
+        copy=True,
+        help="The primary partner contact (If Applicable).",
+    )
+    partner_contact_phone = fields.Char(
+        related="partner_contact_id.phone", string="Partner Phone"
+    )
+    partner_contact_email = fields.Char(
+        related="partner_contact_id.email", string="Partner Email"
+    )
+    company_contact_id = fields.Many2one(
+        "res.partner",
+        string="Company Contact",
+        copy=True,
+        help="The primary contact in the company.",
+    )
+    company_contact_phone = fields.Char(
+        related="company_contact_id.phone", string="Phone"
+    )
+    company_contact_email = fields.Char(
+        related="company_contact_id.email", string="Email"
+    )
+    use_parties_content = fields.Boolean(
+        string="Use parties content", help="Use custom content for parties"
+    )
+    company_partner_id = fields.Many2one(
+        related="company_id.partner_id", string="Company's Partner"
+    )
+
+    def _get_default_parties(self):
+        deftext = """
+        <h3>Company Information</h3>
+        <p>
+        ${object.company_id.partner_id.name or ''}.<br>
+        ${object.company_id.partner_id.street or ''} <br>
+        ${object.company_id.partner_id.state_id.code or ''}
+        ${object.company_id.partner_id.zip or ''}
+        ${object.company_id.partner_id.city or ''}<br>
+        ${object.company_id.partner_id.country_id.name or ''}.<br><br>
+        Represented by <b>${object.company_contact_id.name or ''}.</b>
+        </p>
+        <p></p>
+        <h3>Partner Information</h3>
+        <p>
+        ${object.partner_id.name or ''}.<br>
+        ${object.partner_id.street or ''} <br>
+        ${object.partner_id.state_id.code or ''}
+        ${object.partner_id.zip or ''} ${object.partner_id.city or ''}<br>
+        ${object.partner_id.country_id.name or ''}.<br><br>
+        Represented by <b>${object.partner_contact_id.name or ''}.</b>
+        </p>
+        """
+        return deftext
+
+    parties = fields.Html(
+        string="Parties",
+        tracking=True,
+        default=_get_default_parties,
```

Comment:
```suggestion
        default=lambda self: self._get_default_parties(),
```
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#38fields